exonum/src/blockchain/mod.rs

```diff
@@ -183,12 +184,51 @@ impl Blockchain {
         let has_genesis_block = !Schema::new(&self.snapshot())
             .block_hashes_by_height()
             .is_empty();
-        if !has_genesis_block {
+        if has_genesis_block {
+            self.assert_storage_version();
+        } else {
+            self.initialize_metadata();
             self.create_genesis_block(cfg)?;
         }
         Ok(())
     }
 
+    /// Initialized node-local metadata.
+    fn initialize_metadata(&mut self) {
+        let mut fork = self.db.fork();
+        storage::StorageVersion::write_current(&mut fork);
```

Comment:
Nit: Is there a particular reason for using `StorageVersion` in the qualified form here?

Reply:
I chose `write_current` to be static because it does not actually require any state to do what it is supposed to :)